packages/core/src/lib/components/toggle.tsx

```diff
@@ -18,25 +21,48 @@ const Toggle: React.FC<ToggleProps> = ({ children, ...props }) => {
 
   const toggleOpen = useCallback(() => setOpen((prevOpen) => !prevOpen), []);
 
+  let iconElement: React.ReactNode = null;
+  const otherChildren: React.ReactNode[] = [];
+
+  React.Children.forEach(children, (child) => {
+    if (React.isValidElement(child) && child.type === Toggle.Icon) {
+      iconElement = child.props.children;
+    } else {
+      otherChildren.push(child);
+    }
+  });
+
+  if (!iconElement) {
+    iconElement = <div className="notion-toggle-button-arrow" />;
+  }
+
   return (
     <div
       className={`notion-block notion-toggle  ${getColorCss(color)} ${open ? "notion-toggle-open" : ""}`}
       aria-expanded={open}
     >
       <div className="notion-toggle-content">
-        <button onClick={toggleOpen} className="notion-toggle-button">
+        <button onClick={toggleOpen} className={`notion-toggle-button`}>
           <div
-            className={`notion-toggle-button-arrow ${open ? "notion-toggle-button-arrow-opened" : ""}`}
-          />
+            className={`${open ? "notion-toggle-button-opened" : "notion-toggle-button-closed"}`}
+          >
+            {iconElement}
```

Comment:
p3: dont iconElement inside it

iconElement should contains <div className={`${open ... }} />

custom icon can define its ui on opened or closed.

but It depends on 'notion-toggle-button-opened' css

Reply:
1. In response to "custom icon can define its UI on opened or closed" 
does that mean it should accept two children—one for the open state and one for the closed state?

2. For the default Notion toggle, there is a rotation animation based on the open/closed state, and I would like to maintain the current structure for the default toggle. Is that okay?